src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,207 @@
+            if (!IsClosed)
+            {
+                IsClosed = true;
+                _ = _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, closeReason, _readerCancellationSource.Token).ConfigureAwait(false);
+                TransportClosed?.Invoke(this, new TransportClosedEventArgs { CloseReason = closeReason });
+                _readerCancellationSource?.Cancel();
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Close(Exception ex)
+        {
+            System.Diagnostics.Debug.WriteLine(ex);
+            Close(ex.ToString());
+        }
+
+        private async Task ConnectAsync()
+        {
+            await _webSocket.ConnectAsync(new Uri(_wsEndpoint), _readerCancellationSource.Token).ConfigureAwait(false);
+            ScheduleTransportTask(ReceiveAsync, _readerCancellationSource.Token);
+        }
+
+        private void ScheduleTransportTask(Func<CancellationToken, Task> func, CancellationToken cancellationToken)
+            => Task.Factory.StartNew(() => func(cancellationToken), cancellationToken, TaskCreationOptions.LongRunning, TaskScheduler.Current);
+
+        private async Task ReceiveAsync(CancellationToken token)
+        {
+            try
+            {
+                if (_webSocket.State == WebSocketState.Closed)
+                {
+                    Close("Closed");
+                }
+                else
+                {
+                    byte[] buffer = new byte[DefaultBufferSize];
+
+                    while (!token.IsCancellationRequested && _webSocket.State == WebSocketState.Open)
+                    {
+                        WebSocketReceiveResult result;
+                        using (MemoryStream memoryStream = new())
+                        {
+                            do
+                            {
+                                result = await _webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), _readerCancellationSource.Token).ConfigureAwait(false);
+
+                                if (result.MessageType == WebSocketMessageType.Close)
+                                {
+                                    await
+                                        _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, _readerCancellationSource.Token).ConfigureAwait(false);
+                                }
+                                else
+                                {
+#if NETSTANDARD
+#pragma warning disable CA1835 // We can't use ReadOnlyMemory on netstandard
+                                    await memoryStream.WriteAsync(buffer, 0, result.Count, _readerCancellationSource.Token).ConfigureAwait(false);
+#pragma warning restore CA1835
+#else
+                                await memoryStream.WriteAsync(new(buffer, 0, result.Count), _readerCancellationSource.Token).ConfigureAwait(false);
+#endif
```

Comment:
```suggestion
#if NETSTANDARD
                                    await memoryStream.WriteAsync(buffer.Array, 0, result.Count, _readerCancellationSource.Token).ConfigureAwait(false);
#else
                                await memoryStream.WriteAsync(new(buffer.Array, 0, result.Count), _readerCancellationSource.Token).ConfigureAwait(false);
#endif
```

No need for pragmas when you've got conditional compilation + using the new `.Array` byref call, assuming the comment above gets merged too.

Reply:
fixed